### PR TITLE
V1 4 freeze account

### PR DESCRIPTION
**Pull request by AdoAdoAdo**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,10 +7,10 @@ require (
 	github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1
 	github.com/gin-gonic/gin v1.8.1
 	github.com/mitchellh/mapstructure v1.5.0
-	github.com/multiversx/mx-chain-core-go v1.1.31
-	github.com/multiversx/mx-chain-crypto-go v1.2.5
+	github.com/multiversx/mx-chain-core-go v1.1.34-0.20230308105332-12c26500c7f7
```
Comment (iulianpascalau) [inline go.mod:10]: proper release here and on L13

File: mock/world/builtinFunctionsWrapper.go [hunk 2]
```diff
@@ -38,6 +38,7 @@ func NewBuiltinFunctionsWrapper(
 		MapDNSAddresses:                  dnsMap,
 		Marshalizer:                      WorldMarshalizer,
 		Accounts:                         world.AccountsAdapter,
+    GuardedAccountHandler:            world.GuardedAccountHandler,
```
Comment (iulianpascalau) [inline mock/world/builtinFunctionsWrapper.go:41]: indentation error here

File: go.mod (hunk 1 above)
Comment (iulianpascalau) [reply go.mod:10]: done

File: mock/world/builtinFunctionsWrapper.go (hunk 2 above)
Comment (iulianpascalau) [reply mock/world/builtinFunctionsWrapper.go:41]: done

File: vmhost/mock/enableEpochsHandlerStub.go [hunk 3]
```diff
@@ -50,6 +52,14 @@ type EnableEpochsHandlerStub struct {
 	StorageAPICostOptimizationEnableEpochField           uint32
 }
 
+func (stub *EnableEpochsHandlerStub) IsGuardAccountEnabled() bool {
+	return stub.IsGuardAccountEnabledField
+}
+
+func (stub *EnableEpochsHandlerStub) IsSetGuardianEnabled() bool {
```
Comment (ssd04) [inline vmhost/mock/enableEpochsHandlerStub.go:59]: mock comments?

Comment (mariusmihaic) [reply vmhost/mock/enableEpochsHandlerStub.go:59]: :+1:

File: vmhost/mock/enableEpochsHandlerStub.go [hunk 4]
```diff
@@ -50,6 +52,14 @@ type EnableEpochsHandlerStub struct {
 	StorageAPICostOptimizationEnableEpochField           uint32
 }
 
+func (stub *EnableEpochsHandlerStub) IsGuardAccountEnabled() bool {
```
Comment (sasurobert) [inline vmhost/mock/enableEpochsHandlerStub.go:55]: add comments.

File: vmhost/mock/enableEpochsHandlerStub.go (hunk 3 above)
Comment (iulianpascalau) [reply vmhost/mock/enableEpochsHandlerStub.go]: done

File: vmhost/mock/enableEpochsHandlerStub.go (hunk 4 above)
Comment (iulianpascalau) [reply vmhost/mock/enableEpochsHandlerStub.go:56]: done

**Review by sasurobert — DISMISSED** [review]

**Review by ssd04 — DISMISSED** [review]

**Review by mariusmihaic — DISMISSED** [review]

**Review by raduchis — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

**Pull request merged**